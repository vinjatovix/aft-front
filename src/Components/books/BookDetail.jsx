import PropTypes from "prop-types";

export const BookDetail = ({ data }) => {
  const defaultCoverBook =
    "https://boxshot.com/3d-book-cover/how-to-make-a-3d-book-cover-in-photoshop/sample.jpg";

  return (
    <div className="detail">
      <p>{data.name}</p>
      <p className="small">{data.author}</p>
      <img src={data.img || defaultCoverBook} alt="book" />
      <p>{data.description}</p>
    </div>
  );
};

BookDetail.propTypes = {
  data: PropTypes.shape({
    name: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    img: PropTypes.string,
    description: PropTypes.string.isRequired,
  }).isRequired,
};
