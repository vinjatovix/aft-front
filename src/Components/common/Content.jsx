import React from "react";
import PropTypes from "prop-types";

export const Content = ({ children, modals, modalActions, isAdmin }) => {
  return (
    <div className={modals.blur ? "content blur" : "content"}>
      <div className="content_header">
        {isAdmin && (
          <button className="bg-green" onClick={() => modalActions.add()}>
            AÃ±adir
          </button>
        )}
      </div>
      {children}
    </div>
  );
};

Content.propTypes = {
  children: PropTypes.node.isRequired,
  modals: PropTypes.object.isRequired,
  modalActions: PropTypes.object.isRequired,
  isAdmin: PropTypes.bool,
};

Content.defaultProps = {
  isAdmin: false,
};
