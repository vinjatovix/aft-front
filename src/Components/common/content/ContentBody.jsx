import React from "react";
import PropTypes from "prop-types";
import { ListItem } from "../ListItem";
import { ContentListDetail } from "./ContentListDetail";

export const ContentBody = ({ content, isAdmin, actions, token, type }) => {
  return (
    <ul className="content-list">
      {content &&
        content.map((item) => (
          <ListItem key={item._id} className="content-list_item" {...item}>
            <ContentListDetail
              type={type}
              isAdmin={isAdmin}
              token={token}
              actions={actions}
              item={item}
            />
          </ListItem>
        ))}
    </ul>
  );
};

ContentBody.propTypes = {
  content: PropTypes.array,
  isAdmin: PropTypes.bool,
  actions: PropTypes.object.isRequired,
  token: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
};

ContentBody.defaultProps = {
  content: [],
  isAdmin: false,
};
