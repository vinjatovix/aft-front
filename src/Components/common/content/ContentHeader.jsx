import React from "react";
import PropTypes from "prop-types";

export const ContentHeader = ({ book, title, count }) => {
  return (
    <>
      {book && (
        <>
          <img
            className="portrait"
            src={
              book.img
                ? book.img
                : "https://boxshot.com/3d-book-cover/how-to-make-a-3d-book-cover-in-photoshop/sample.jpg"
            }
            alt="book"
          />
          <h1>
            {book.author}-{book.name}
          </h1>
          <h2>{book.description}</h2>
        </>
      )}

      <h3 data-testid={title}>
        {title} {count}
      </h3>
    </>
  );
};

ContentHeader.propTypes = {
  book: PropTypes.shape({
    name: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    img: PropTypes.string,
    description: PropTypes.string.isRequired,
  }),
  title: PropTypes.string.isRequired,
  count: PropTypes.number,
};

ContentHeader.defaultProps = {
  book: null,
  count: null,
};
