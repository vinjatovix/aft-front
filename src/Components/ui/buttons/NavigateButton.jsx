import React from "react";
import PropTypes from "prop-types";

export const NavigateButton = ({ text }) => {
  return (
    <button data-testid="navigate-button" className="bg-blue">
      {text}
    </button>
  );
};

NavigateButton.propTypes = {
  text: PropTypes.string,
};

NavigateButton.defaultProps = {
  text: "➡️ Go",
};
