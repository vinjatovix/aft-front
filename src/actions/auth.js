import { ACTION_TYPES } from "../types/types";

export const loginRequestAction = () => ({
  type: ACTION_TYPES.AUTH_LOGIN_REQUEST,
});

export const loginSuccessAction = (auth) => ({
  type: ACTION_TYPES.AUTH_LOGIN_SUCCESS,
  payload: auth,
});

export const loginFailureAction = (error) => ({
  type: ACTION_TYPES.AUTH_LOGIN_FAILURE,
  payload: error,
});

export const logoutAction = () => ({
  type: ACTION_TYPES.AUTH_LOGOUT,
});
