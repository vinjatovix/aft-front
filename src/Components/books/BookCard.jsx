import React from "react";
import PropTypes from "prop-types";
import { NavLink } from "react-router-dom";

import { ListItem } from "../common/ListItem";
import { NavigateButton } from "../ui/buttons/NavigateButton";
import { AdminButtons } from "../ui/admin/AdminButtons";

const DEFAULT_COVER_BOOK =
  "https://boxshot.com/3d-book-cover/how-to-make-a-3d-book-cover-in-photoshop/sample.jpg";

export const BookCard = ({ data, isEditor, actions, token }) => (
  <div className="flex-wrapper column">
    <ul
      data-testid="book-detail"
      className="book"
      onClick={() => {
        actions.detail();
      }}
    >
      {<img src={data?.img || DEFAULT_COVER_BOOK} alt={data.name} />}
      <ListItem name="Obra" children={data.name} />
      <ListItem name="Autor" className="small" children={data.author} />
    </ul>

    <NavLink to={`/book/${data._id}/characters`}>
      <NavigateButton text="ðŸŽ­ Personajes" />
    </NavLink>

    <NavLink to={`/book/${data._id}/scenes`}>
      <NavigateButton text="ðŸŽ¬ Escenas" />
    </NavLink>

    {isEditor && <AdminButtons token={token} item={data} actions={actions} />}
  </div>
);

BookCard.propTypes = {
  data: PropTypes.shape({
    name: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    img: PropTypes.string,
    description: PropTypes.string.isRequired,
  }).isRequired,
  isEditor: PropTypes.bool,
  actions: PropTypes.shape({
    setDataDetail: PropTypes.func.isRequired,
    add: PropTypes.func.isRequired,
    close: PropTypes.func.isRequired,
    delete: PropTypes.func.isRequired,
    edit: PropTypes.func.isRequired,
    detail: PropTypes.func.isRequired,
    refresh: PropTypes.func.isRequired,
  }).isRequired,
  token: PropTypes.string.isRequired,
};

BookCard.defaultProps = {
  isEditor: false,
};
