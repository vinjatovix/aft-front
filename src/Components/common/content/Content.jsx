import React from "react";
import PropTypes from "prop-types";
import { ActionButton } from "../../ui/buttons/ActionButton";
import { modalsStateType } from "../../../PropTypes";

export const Content = ({ children, modals, actions, isEditor, isOpen }) => {
  return (
    <div className={modals.blur ? "content blur" : "content"}>
      <div className="content_header">
        {(isEditor || isOpen) && (
          <ActionButton text="AÃ±adir" type="bg-green" action={actions.add} />
        )}
      </div>
      {children}
    </div>
  );
};

Content.propTypes = {
  children: PropTypes.node.isRequired,
  modals: modalsStateType,
  actions: PropTypes.shape({
    add: PropTypes.func.isRequired,
  }).isRequired,
  isEditor: PropTypes.bool,
};

Content.defaultProps = {
  isEditor: false,
};
