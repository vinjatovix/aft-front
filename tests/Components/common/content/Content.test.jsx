import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

import React from "react";
import { BrowserRouter as Router, NavLink } from "react-router-dom";

import { Content } from "../../../../src/Components/common/content/Content";
import { NavigateButton } from "../../../../src/Components/ui/buttons/NavigateButton";
import { ContentHeader } from "../../../../src/Components/common/content/ContentHeader";
import { ContentBody } from "../../../../src/Components/common/content/ContentBody";
import actions from "../../../fixtures/actions";
import books from "../../../fixtures/books";
import scenes from "../../../fixtures/scenes";
import characters from "../../../fixtures/characters";

describe("Content Component", () => {
  it("should render scenes content", async () => {
    const Entity = { data: scenes };

    const dispatchEntity = { count: 3 };
    const isAdmin = true;
    const auth = { token: "token" };
    const modals = {
      add: false,
      blur: false,
      delete: false,
      detail: false,
      edit: false,
      data: {},
    };

    const { container } = render(
      <Router>
        <Content modals={modals} modalActions={actions} isAdmin={isAdmin}>
          <NavLink to={`/characters/book/${books[0]._id}`}>
            <NavigateButton text="ðŸŽ­ Personajes" />
          </NavLink>

          <ContentHeader
            book={books[0]}
            title="Escenas:"
            count={Entity.data.length}
          />

          <ContentBody
            type="scene"
            content={Entity.data}
            isAdmin={isAdmin}
            actions={actions}
            token={auth.token}
          />
        </Content>
      </Router>
    );

    expect(container).toMatchSnapshot();
  });

  it("should render characters content", async () => {
    const Entity = { data: characters };

    const isAdmin = true;
    const auth = { token: "token" };
    const modals = {
      add: false,
      blur: false,
      delete: false,
      detail: false,
      edit: false,
      data: {},
    };

    const { container } = render(
      <Router>
        <Content modals={modals} modalActions={actions} isAdmin={isAdmin}>
          <NavLink to={`/characters/book/${books[0]._id}`}>
            <NavigateButton text="ðŸŽ­ Personajes" />
          </NavLink>

          <ContentHeader
            book={books[0]}
            title="Personajes:"
            count={Entity.data.length}
          />

          <ContentBody
            type="character"
            content={Entity.data}
            isAdmin={isAdmin}
            actions={actions}
            token={auth.token}
          />
        </Content>
      </Router>
    );

    expect(container).toMatchSnapshot();
  });
});
