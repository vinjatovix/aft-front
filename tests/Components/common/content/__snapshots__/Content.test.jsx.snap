// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Component should render characters content 1`] = `
<div>
  <div
    class="content"
  >
    <div
      class="content_header"
    >
      <button
        class="bg-green"
        data-testid="action-button"
      >
        Añadir
      </button>
    </div>
    <a
      class=""
      href="/characters/book/6301216103b6fbb712bbde7c"
    >
      <button
        class="bg-blue"
        data-testid="navigate-button"
      >
        🎭 Personajes
      </button>
    </a>
    <div
      class="content-header"
    >
      <h1
        data-testid="Personajes:"
      >
        Personajes:
         
        3
      </h1>
      <img
        alt="book"
        class="portrait"
        src="https://boxshot.com/3d-book-cover/how-to-make-a-3d-book-cover-in-photoshop/sample.jpg"
      />
      <h2>
        Theresa Larkin
         - 
        transmitter Refined
      </h2>
      <h6>
        I'll index the auxiliary SAS program, that should interface the USB driver!
      </h6>
    </div>
    <ul
      class="content-list"
    >
      <li
        class="content-list_item"
        data-testid="Agnes Kris"
      >
        <div>
          <p>
            Agnes Kris
            : 
            encompassing
            . 
            instintive
          </p>
        </div>
        <div
          class="control-panel-mini row"
        >
          <button
            class="btn-big bg-yellow"
            data-testid="action-button"
          >
            📝
          </button>
          <button
            class="btn-big bg-red"
            data-testid="action-button"
          >
            🗑️
          </button>
        </div>
      </li>
      <li
        class="content-list_item"
        data-testid="Leon Mohr"
      >
        <div>
          <p>
            Leon Mohr
            : 
            background
            . 
            instintive
          </p>
        </div>
        <div
          class="control-panel-mini row"
        >
          <button
            class="btn-big bg-yellow"
            data-testid="action-button"
          >
            📝
          </button>
          <button
            class="btn-big bg-red"
            data-testid="action-button"
          >
            🗑️
          </button>
        </div>
      </li>
      <li
        class="content-list_item"
        data-testid="Terrence Rosenbaum"
      >
        <div>
          <p>
            Terrence Rosenbaum
            : 
            Global
            . 
            instintive
          </p>
        </div>
        <div
          class="control-panel-mini row"
        >
          <button
            class="btn-big bg-yellow"
            data-testid="action-button"
          >
            📝
          </button>
          <button
            class="btn-big bg-red"
            data-testid="action-button"
          >
            🗑️
          </button>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Content Component should render scenes content 1`] = `
<div>
  <div
    class="content"
  >
    <div
      class="content_header"
    >
      <button
        class="bg-green"
        data-testid="action-button"
      >
        Añadir
      </button>
    </div>
    <a
      class=""
      href="/characters/book/6301216103b6fbb712bbde7c"
    >
      <button
        class="bg-blue"
        data-testid="navigate-button"
      >
        🎭 Personajes
      </button>
    </a>
    <div
      class="content-header"
    >
      <h1
        data-testid="Escenas:"
      >
        Escenas:
         
        4
      </h1>
      <img
        alt="book"
        class="portrait"
        src="https://boxshot.com/3d-book-cover/how-to-make-a-3d-book-cover-in-photoshop/sample.jpg"
      />
      <h2>
        Theresa Larkin
         - 
        transmitter Refined
      </h2>
      <h6>
        I'll index the auxiliary SAS program, that should interface the USB driver!
      </h6>
    </div>
    <ul
      class="content-list"
    >
      <li
        class="content-list_item"
        data-testid="e - User-centric"
      >
        <div>
          <p>
            e - User-centric
            -
            Predovichaven
             
            Sun Oct 03 2021 03:16:19 GMT+0200 (hora de verano de Europa central)
            : 
            programming the card won't do anything, we need to hack the solid state AGP driver!
            .
          </p>
        </div>
        <div
          class="control-panel-mini row"
        >
          <button
            class="btn-big bg-yellow"
            data-testid="action-button"
          >
            📝
          </button>
          <button
            class="btn-big bg-red"
            data-testid="action-button"
          >
            🗑️
          </button>
        </div>
      </li>
      <li
        class="content-list_item"
        data-testid="l - Centralized"
      >
        <div>
          <p>
            l - Centralized
            -
            Lomahaven
             
            Mon Jan 17 2022 13:41:10 GMT+0100 (hora estándar de Europa central)
            : 
            If we index the circuit, we can get to the XSS matrix through the bluetooth PNG bus!
            .
          </p>
        </div>
        <div
          class="control-panel-mini row"
        >
          <button
            class="btn-big bg-yellow"
            data-testid="action-button"
          >
            📝
          </button>
          <button
            class="btn-big bg-red"
            data-testid="action-button"
          >
            🗑️
          </button>
        </div>
      </li>
      <li
        class="content-list_item"
        data-testid="y - Synergized"
      >
        <div>
          <p>
            y - Synergized
            -
            Aufderharbury
             
            Sat Oct 16 2021 22:26:24 GMT+0200 (hora de verano de Europa central)
            : 
            The TCP array is down, transmit the digital bandwidth so we can reboot the HDD pixel!
            .
          </p>
        </div>
        <div
          class="control-panel-mini row"
        >
          <button
            class="btn-big bg-yellow"
            data-testid="action-button"
          >
            📝
          </button>
          <button
            class="btn-big bg-red"
            data-testid="action-button"
          >
            🗑️
          </button>
        </div>
      </li>
      <li
        class="content-list_item"
        data-testid="y - Progressive"
      >
        <div>
          <p>
            y - Progressive
            -
            New Theronfort
             
            Sun Nov 07 2021 01:37:29 GMT+0100 (hora estándar de Europa central)
            : 
            Use the multi-byte HDD port, then you can compress the bluetooth matrix!
            .
          </p>
        </div>
        <div
          class="control-panel-mini row"
        >
          <button
            class="btn-big bg-yellow"
            data-testid="action-button"
          >
            📝
          </button>
          <button
            class="btn-big bg-red"
            data-testid="action-button"
          >
            🗑️
          </button>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
